### PR TITLE
feat: add story's notification on feed page after sharing story

### PR DESCRIPTION
**Pull request by ice-alcides**
Figma [link](https://www.figma.com/design/3TnZngoFeklOVC8TexlR1D/%F0%9F%A7%8A-ICE-Cloud?node-id=41881-109053&node-type=frame&t=G4UwLnFqmDvXeA4y-0)

- add notification on the `FeedPage` after sharing a story with an animation's effects
- refactor `use_camera_recording_controller`
- rename `StoryCameraRoute` to `StoryRecordPage`
- add `ContentNotificationBar` to handle the different types of notifications and states
- fix `visibility_settings_list_item.dart `to have a possibility to select some option

### CODE REVIEW

File: lib/app/features/feed/create_story/views/components/story_video/story_published_notification.dart [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      child: SizedBox(
+        height: height,
+        child: ColoredBox(
+          color: context.theme.appColors.primaryAccent,
+          child: Padding(
+            padding: EdgeInsets.symmetric(vertical: 3.0.s),
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: [
+                Assets.svg.iconBlockCheckboxwhiteOn.icon(
```
[the remark this answers is not in the document]
Comment (ice-alcides) [reply lib/app/features/feed/create_story/views/components/story_video/story_published_notification.dart]: thanks, changed to` icon/block_checkbox_onblue`

File: lib/app/features/feed/create_story/views/components/story_video/story_published_notification.dart [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      storyCameraControllerProvider.select((state) => state.isStoryPublished),
+    );
+
+    return Animate(
+      effects: topSlideInOutEffects,
+      onComplete: (controller) =>
+          ref.read(storyCameraControllerProvider.notifier).resetStoryPublished(),
+      autoPlay: isStoryPublished,
+      child: SizedBox(
+        height: height,
```
Comment (ice-kreios) [inline lib/app/features/feed/create_story/views/components/story_video/story_published_notification.dart:28]: If the i18n text is long and might require multiple lines for different languages, we should ensure the layout is flexible enough to accommodate multiline text.

Comment (ice-alcides) [reply lib/app/features/feed/create_story/views/components/story_video/story_published_notification.dart:28]: yeah, we should test the app entirely to ensure the layouts are fine in all places

Comment (ice-alcides) [reply lib/app/features/feed/create_story/views/components/story_video/story_published_notification.dart:28]: @ice-orion  @ice-hades A fixed height is needed to move content below the notification on the Feed Page and avoid overlapping the notification with the content. Wrapping with a `Column` is not necessary in this case.

File: lib/app/features/feed/create_story/views/components/story_video/story_published_notification.dart [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+import 'package:flutter/material.dart';
+import 'package:flutter_animate/flutter_animate.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:ion/app/effects/slide_in_out_effects.dart';
+import 'package:ion/app/extensions/extensions.dart';
+import 'package:ion/app/features/feed/create_story/providers/story_camera_provider.dart';
+import 'package:ion/generated/assets.gen.dart';
+
+class StoryPublishedNotification extends ConsumerWidget {
```
[the remark this answers is not in the document]
Comment (ice-alcides) [reply lib/app/features/feed/create_story/views/components/story_video/story_published_notification.dart:11]: Do you propose to do it in this PR? There is a to-do comment in FeedPage: `// TODO: refactor to support other types of notifications`.

File: lib/app/features/feed/create_story/views/pages/story_record_page.dart [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+import 'package:ion/app/router/app_routes.dart';
+
+class StoryRecordPage extends HookConsumerWidget {
+  const StoryRecordPage({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final storyCameraState = ref.watch(storyCameraControllerProvider);
+
+    useOnAppLifecycleStateChange((_, AppLifecycleState current) async {
```
Comment (ice-tychon) [inline lib/app/features/feed/create_story/views/pages/story_record_page.dart:27]: We have a `AppLifecycleProvider` now, I suggest you to migrate and use it inside `cameraControllerNotifierProvider`. The issue is that you might forget to pause/resume camera if you use the provider in multiple places.

Comment (ice-alcides) [reply lib/app/features/feed/create_story/views/pages/story_record_page.dart:27]: yeah, makes sense

File: lib/app/effects/slide_in_out_effects.dart [hunk 5]
```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:flutter/material.dart';
+import 'package:flutter_animate/flutter_animate.dart';
+
+final List<Effect<dynamic>> topSlideInOutEffects = [
```
[the remark this answers is not in the document]
Comment (ice-alcides) [reply lib/app/effects/slide_in_out_effects.dart:6]: done

File: lib/app/features/feed/create_story/views/pages/story_record_page.dart (hunk 4 above)
Comment (ice-alcides) [reply lib/app/features/feed/create_story/views/pages/story_record_page.dart]: done

**Review by ice-kreios — APPROVED** [review]

**Review by ice-tychon — APPROVED** [review]

**Pull request merged**